spec/models/manageiq/providers/cloud_manager/provision_workflow_spec.rb

```diff
@@ -35,7 +35,7 @@
 
       result = workflow.allowed_customization_templates(options)
       customization_template = workflow.instance_variable_get(:@values)[:customization_template_script]
-      template_hash = result.first.instance_variable_get(:@hash)
+      template_hash = result.first.instance_variable_get(:@table)
```

Comment:
Note:  This change and the one below it are because the previous version of the spec is grabbing the internal variable inside the `MiqHashStruct`, and using that to test the resulting values.  `@hash` is the internal hash variable in `MiqHashStruct`, and `@table` is the equivalent in `OpenStruct`.

I personally think it make more sense to avoid the `instance_variable_get` all together, and just do a `send(:attr)` instead, but I didn't want to change the tests that I hardly understand in the first place.  They were added recent, however, in these two PRs:

- https://github.com/ManageIQ/manageiq/pull/17293
- https://github.com/ManageIQ/manageiq/pull/17477

Reply:
You can just call `result.first.to_h` if all you want is the hash.  That works equivalently in MiqHashStruct and OpenStruct.